contrib/platform/src/com/sun/jna/platform/win32/WinGDI.java

```diff
@@ -28,18 +28,97 @@
 import com.sun.jna.Structure;
 import com.sun.jna.Structure.FieldOrder;
 import com.sun.jna.platform.win32.WinNT.HANDLE;
-import com.sun.jna.platform.win32.WinDef.HBITMAP;
-import com.sun.jna.platform.win32.WinDef.RECT;
+import com.sun.jna.Union;
+
+import static com.sun.jna.platform.win32.WinDef.*;
 
 /**
  * Ported from WinGDI.h.
  * Microsoft Windows SDK 6.0A.
  * @author dblock[at]dblock.org
  * @author Andreas "PAX" L&uuml;ck, onkelpax-git[at]yahoo.de
+ * @author SSHTOOLS Limited, support@sshtools.com
  */
 public interface WinGDI {
     int RDH_RECTANGLES = 1;
 
+    @FieldOrder({ "dmDeviceName", "dmSpecVersion", "dmDriverVersion", "dmSize", "dmDriverExtra", "dmFields", "dmUnion1", "dmColor",
+            "dmDuplex", "dmYResolution", "dmTTOption", "dmCollate", "dmFormName", "dmLogPixels", "dmBitsPerPel", "dmPelsWidth",
+            "dmPelsHeight", "dummyunionname2", "dmDisplayFrequency", "dmICMMethod", "dmICMIntent", "dmMediaType", "dmDitherType",
+            "dmReserved1", "dmReserved2", "dmPanningWidth", "dmPanningHeight" })
+
+    public static class DEVMODE extends Structure {
+        public byte[] dmDeviceName = new byte[Winspool.CCHDEVICENAME];
```

Comment:
I'm having difficulty converting the `byte[]` returned from `dmDeviceName` into a usable Java string, it prints gibberish and the gibberish changes each time I invoke the API.

Reply:
See above.  You have the choice of defining `DEVMODEW` and making that a `char[]` field, and using that, or using `DEVMODE` and handling both cases.

Reply:
Resolved via https://github.com/java-native-access/jna/pull/1429#discussion_r837838279.